Library/Homebrew/cmd/audit.rb

```diff
@@ -747,6 +748,20 @@ def audit_caveats
     end
   end
 
+  def audit_reverse_migration
+    # Only enforce for new formula being re-added to core
+    return unless @strict
+    return unless formula.tap.downcase == "homebrew/homebrew"
+
+    unless TAP_MIGRATIONS.map(&:to_s).grep(/#{formula.name}/).empty?
```

Comment:
On second thought, string matching is incorrect. It will return false result on partial matching.